src/pages/api/breadcrumbs.ts

```diff
@@ -135,54 +125,122 @@ async function fetchLabel(
       const campaign = await apiFetch(
         `/orgs/${orgId}/campaigns/${fieldValue}`
       ).then((res) => res.json());
-      return campaign.data.title;
+      return {
+        href: basePath + '/' + fieldValue,
+        label: campaign.data.title,
+      };
     }
   }
   if (fieldName === 'taskId') {
     const task = await apiFetch(`/orgs/${orgId}/tasks/${fieldValue}`).then(
       (res) => res.json()
     );
-    return task.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: task.data.title,
+    };
   }
   if (fieldName == 'viewId') {
     const view = await apiFetch(
       `/orgs/${orgId}/people/views/${fieldValue}`
     ).then((res) => res.json());
-    return view.data.title;
+    return {
+      folderId: view.data?.folder?.id,
+      href: basePath + '/' + fieldValue,
+      label: view.data.title,
+    };
   }
   if (fieldName == 'instanceId') {
     const journeyInstance = await apiFetch(
       `/orgs/${orgId}/journey_instances/${fieldValue}`
     ).then((res) => res.json());
-    return `${
-      journeyInstance.data.title || journeyInstance.data.journey.title
-    } #${journeyInstance.data.id}`;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: `${
+        journeyInstance.data.title || journeyInstance.data.journey.title
+      } #${journeyInstance.data.id}`,
+    };
   }
   if (fieldName == 'journeyId') {
     const journey = await apiFetch(
       `/orgs/${orgId}/journeys/${fieldValue}`
     ).then((res) => res.json());
-    return journey.data.plural_label;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: journey.data.plural_label,
+    };
   }
   if (fieldName == 'callAssId') {
     const assignment = await apiFetch(
       `/orgs/${orgId}/call_assignments/${fieldValue}`
     ).then((res) => res.json());
-    return assignment.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: assignment.data.title,
+    };
   }
   if (fieldName == 'surveyId') {
     const survey = await apiFetch(`/orgs/${orgId}/surveys/${fieldValue}`).then(
       (res) => res.json()
     );
-    return survey.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: survey.data.title,
+    };
   }
   if (fieldName == 'eventId') {
     const event = await apiFetch(`/orgs/${orgId}/actions/${fieldValue}`).then(
       (res) => res.json()
     );
-    return event.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: event.data.title,
+    };
   }
-  return fieldValue;
+  if (fieldName == 'folderId') {
+    const folders = await apiFetch(`/orgs/${orgId}/people/view_folders`)
+      .then((res) => res.json())
+      .then((envelope) => envelope.data as ZetkinViewFolder[]);
+
+    const folder = folders.find((folder) => folder.id == parseInt(fieldValue));
+
+    if (folder) {
```

Comment:
Wouldn't it be possible to expand the stack here, and return multiple folders in one go. I realize this function would need to change a little bit for that to work, but it still seems to me like a smaller change than the one made here.

It could be done by `yield` ing (converting the function to a generator) or by always returning an array of elements even if most of the branches in this function will only return a single-element array.

Reply:
I went with always returning an array in the latest revision.